src/sqlfluff/core/rules/base.py

```diff
@@ -614,6 +684,12 @@ class BaseRule(metaclass=RuleMetaclass):
     # referred to this rule.
     aliases: Tuple[str, ...] = ()
 
+    # NOTE: code and description are provided here as hints, but should not
+    # be set directly. They are set automatically by the metaclass based on
+    # the class _name_ when defined.
+    code: str = ""
```

Comment:
for hints, do you need to set a default value? Can't you simply
```
code: str
```

Reply:
Good question - I'll try it 🤷

Reply:
looks like that works. Change coming through now...